ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -57,111 +53,104 @@
 
-  public String defaultAutoOffsetRestConfig = "latest";
-  public long defaultCommitIntervalMsConfig = 2000;
-  public long defaultCacheMaxBytesBufferingConfig = 10000000;
-  public int defaultNumberOfStreamsThreads = 4;
+  private static String defaultAutoOffsetRestConfig = "latest";
+  private static long defaultCommitIntervalMsConfig = 2000;
+  private static long defaultCacheMaxBytesBufferingConfig = 10000000;
+  private static int defaultNumberOfStreamsThreads = 4;
 
   Map<String, Object> ksqlConfigProps;
   Map<String, Object> ksqlStreamConfigProps;
 
-  private static final ConfigDef CONFIG_DEF = new ConfigDef();
+  private static final ConfigDef CONFIG_DEF;
+
+  static {
+    CONFIG_DEF = new ConfigDef()
+    .define(KSQL_SERVICE_ID_CONFIG,
+            ConfigDef.Type.STRING,
+            KSQL_SERVICE_ID_DEFAULT,
+            ConfigDef.Importance.MEDIUM,
+            "Indicates the ID of the ksql service. It will be used as prefix for all KSQL queries in "
+            + "this service.")
+
+    .define(KSQL_PERSISTENT_QUERY_NAME_PREFIX_CONFIG,
+            ConfigDef.Type.STRING,
+            KSQL_PERSISTENT_QUERY_NAME_PREFIX_DEFAULT,
+            ConfigDef.Importance.MEDIUM,
+            "Second part of the prefix for persitent queries. For instance if the prefix is transient_"
+            + "query_ the query name will be ksql_query_1.")
+    .define(KSQL_TRANSIENT_QUERY_NAME_PREFIX_CONFIG,
+            ConfigDef.Type.STRING,
+            KSQL_TRANSIENT_QUERY_NAME_PREFIX_DEFAULT,
+            ConfigDef.Importance.MEDIUM,
+            "Second part of the prefix for transient queries. For instance if the prefix is "
+            + "transient_ the query name would be ksql_transient_4120896722607083946_1509389010601")
+    .define(KSQL_TABLE_STATESTORE_NAME_SUFFIX_CONFIG,
+            ConfigDef.Type.STRING,
+            KSQL_TABLE_STATESTORE_NAME_SUFFIX_DEFAULT,
+            ConfigDef.Importance.MEDIUM,
+            "Suffix for state store names in Tables. For instance if the suffix is _ksql_statestore the state "
+            + "store name would be ksql_query_1_ksql_statestore"
+            + "_ksql_statestore ")
+    .define(SINK_NUMBER_OF_PARTITIONS_PROPERTY,
+            ConfigDef.Type.INT,
+            defaultSinkNumberOfPartitions,
+            ConfigDef.Importance.MEDIUM,
+            "The default number of partitions for the topics created by KSQL.")
+    .define(SINK_NUMBER_OF_REPLICAS_PROPERTY,
+            ConfigDef.Type.SHORT,
+            defaultSinkNumberOfReplications,
+            ConfigDef.Importance.MEDIUM,
+            "The default number of replicas for the topics created by KSQL."
+            )
+    .define(SINK_WINDOW_CHANGE_LOG_ADDITIONAL_RETENTION_PROPERTY,
+            ConfigDef.Type.LONG,
+            defaultSinkWindowChangeLogAdditionalRetention,
+            ConfigDef.Importance.MEDIUM,
+            "The default window change log additional retention time."
```

Comment:
You probably want to drop the 'default' here. Also, it may make sense to expand the description. It doesn't make sense as is. Perhaps a sentence like 'The amount of time to retain window change logs'. Actually I am still not sure what that means Also, there should be a time unit in the name of the config and the description.

Reply:
Good point, added the time unit and more details to the description.